src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+# Copyright 2024 Marigold authors, PRS ETH Zurich. All rights reserved.
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# --------------------------------------------------------------------------
+# More information and citation instructions are available on the
+# Marigold project website: https://marigoldmonodepth.github.io
+# --------------------------------------------------------------------------
+import json
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from PIL import Image
+from scipy.optimize import minimize
+from tqdm.auto import tqdm
+from transformers import CLIPTextModel, CLIPTokenizer
+
+from ...image_processor import PipelineImageInput
+from ...models import (
+    AutoencoderKL,
+    UNet2DConditionModel,
+)
+from ...schedulers import (
+    DDIMScheduler,
+    LCMScheduler,
+)
+from ...utils import (
+    BaseOutput,
+    deprecate,
+    logging,
+    replace_example_docstring,
+)
+from ...utils.torch_utils import randn_tensor
+from ..pipeline_utils import DiffusionPipeline
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+BIBTEX = """
+```bibtex
+@InProceedings{ke2023repurposing,
+  title={Repurposing Diffusion-Based Image Generators for Monocular Depth Estimation},
+  author={Bingxin Ke and Anton Obukhov and Shengyu Huang and Nando Metzger and Rodrigo Caye Daudt and Konrad Schindler},
+  booktitle = {Proceedings of the IEEE/CVF Conference on Computer Vision and Pattern Recognition (CVPR)},
+  year={2024}
+}
+```
+"""
+
+EXAMPLE_DOC_STRING = f"""
+Examples:
+```py
+>>> import requests
+>>> from diffusers import MarigoldDepthPipeline
+>>> from PIL import Image
+
+>>> pipe = MarigoldDepthPipeline.from_pretrained("prs-eth/marigold-depth-lcm-v1-0")
+>>> pipe = pipe.to("cuda")
+
+>>> image = Image.open(requests.get("https://marigoldmonodepth.github.io/images/einstein.jpg", stream=True).raw)
+>>> depth = pipe(image, preset="fast", output_visualization=True)
+
+>>> depth.visualization.save("einstein_depth.png")
+```
+
+Citation: {BIBTEX}
+"""
+
+
+def resize_maybe_antialias(image: torch.Tensor, size: Tuple[int, int], mode: str, is_aa: bool = None) -> torch.Tensor:
```

Comment:
(nit): slightly prefer `maybe_resize_antialias`

Reply:
Also cc: @yiyixuxu here. Should these image processing related utilities (such as `resize_maybe_antialias()`, `resize_to_max_edge()`, etc.) reside in an `MariGoldImageProcessor` class, rather? 

Good example:
https://github.com/huggingface/diffusers/blob/0d7c4790235ac00b4524b492bc2a680dcc5cf6b0/src/diffusers/image_processor.py#L999

Reply:
I have created `MarigoldImageProcessor` now. I subclassed it from `ConfigMixin` and tried to follow the design of `VaeImageProcessor`. I extracted all common parts from the depth and normals pipelines into the processor. Resolving this for now.